qubesmanager/restore.py

```diff
@@ -126,119 +103,102 @@ def __fill_vms_list__(self):
+
+            if self.ignore_uname_mismatch.isChecked():
+                self.backup_restore.options.ignore_username_mismatch = True
+
+            if self.verify_only.isChecked():
+                self.backup_restore.options.verify_only = True
+
+            self.vms_to_restore = self.backup_restore.get_restore_info()
 
             for vmname in self.vms_to_restore:
                 if vmname.startswith('$'):
                     # Internal info
                     continue
                 self.select_vms_widget.available_list.addItem(vmname)
-        except QubesException as ex:
-            QMessageBox.warning (None, self.tr("Restore error!"), str(ex))
-
-    def __init_restore_options__(self):
-        if not self.restore_options:
-            self.restore_options = {}
-            backup.backup_restore_set_defaults(self.restore_options)
+        except exc.QubesException as ex:
+            QtGui.QMessageBox.warning(None, self.tr("Restore error!"), str(ex))
 
-        if 'use-default-template' in self.restore_options and 'use-default-netvm' in self.restore_options:
-            val = self.restore_options['use-default-template'] and self.restore_options['use-default-netvm']
-            self.ignore_missing.setChecked(val)
-        else:
-            self.ignore_missing.setChecked(False)
-
-        if 'ignore-username-mismatch' in self.restore_options:
-            self.ignore_uname_mismatch.setChecked(self.restore_options['ignore-username-mismatch'])
-
-    def gather_output(self, s):
-        self.func_output.append(s)
+    def append_output(self, text):
+        self.commit_text_edit.append(text)
 
-    def restore_error_output(self, s):
+    def restore_error_output(self, text):
         self.error_detected.set()
-        self.feedback_queue.put((SIGNAL("restore_progress(QString)"),
-                                 u'<font color="red">{0}</font>'.format(s)))
+        self.append_output(u'<font color="red">{0}</font>'.format(text))
 
-    def restore_output(self, s):
-        self.feedback_queue.put((SIGNAL("restore_progress(QString)"),
-                                 u'<font color="black">{0}</font>'.format(s)))
+    def restore_output(self, text):
+        self.append_output(u'<font color="black">{0}</font>'.format(text))
 
-    def update_progress_bar(self, value):
-        self.feedback_queue.put((SIGNAL("backup_progress(int)"), value))
-
-    def __do_restore__(self, thread_monitor):
+    def __do_restore__(self, t_monitor):
         err_msg = []
-        self.qvm_collection.lock_db_for_writing()
         try:
-            backup.backup_restore_do(self.vms_to_restore,
-                                     self.qvm_collection,
-                                     print_callback=self.restore_output,
-                                     error_callback=self.restore_error_output,
-                                     progress_callback=self.update_progress_bar)
+            self.backup_restore.restore_do(self.vms_to_restore)
```

Comment:
New `BackupRestore.backup_do` use standard python logging instead of custom callbacks. You should register logger for `qubesadmin.backup` to catch those messages and display them. See here https://docs.python.org/3/library/logging.handlers.html#queuehandler
Then set it like this:
```python
handler = logging.handlers.QueueHandler(self.feedback_queue)
logger = logging.getLogger('qubesadmin.backup')
logger.addHandler(handler)
```
Handling elements in `self.feedback_queue` (in `current_page_changed`) may need to be adjusted for the format used by QueueHandler.